pkg/controllers/customcluster_controller.go

```diff
@@ -250,6 +293,111 @@ func (r *CustomClusterController) reconcileHandleProvisioning(ctx context.Contex
 	return ctrl.Result{}, nil
 }
 
+// reconcileScaleUp is responsible for handling the customCluster reconciliation process when worker nodes need to be scaled up.
+func (r *CustomClusterController) reconcileScaleUp(ctx context.Context, customCluster *v1alpha1.CustomCluster, scaleUpWorkerNodes []NodeInfo) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Checks whether the worker node for scaling up already exists. If it does not exist, the function creates it.
+	workerPod, errGetPod := r.ensureWorkerPodCreated(ctx, customCluster, CustomClusterScaleUpAction, KubesprayScaleUpCMD)
+	if errGetPod != nil {
+		log.Error(errGetPod, "failed to ensure that scaleUp WorkerPod is created ", "customCluster", customCluster.Name)
+		return ctrl.Result{RequeueAfter: RequeueAfter}, errGetPod
+	}
+
+	// Check the current customCluster status, if it is "Provisioned", then change it to "ScalingUp".
+	if customCluster.Status.Phase == v1alpha1.ProvisionedPhase {
```

Comment:
can check customCluster.Status.Phase != v1alpha1.ScalingUpPhase since the `ProvisionedPhase` has been guranteed before